tests/disruption_helpers.py

```diff
@@ -9,65 +9,54 @@
-            self.resource_obj = pod.get_mgr_pods()
-            self.selector = constants.MGR_APP_LABEL
-        if self.resource == 'mon':
-            self.resource_obj = pod.get_mon_pods()
-            self.selector = constants.MON_APP_LABEL
-        if self.resource == 'osd':
-            self.resource_obj = pod.get_osd_pods()
-            self.selector = constants.OSD_APP_LABEL
-        if self.resource == 'mds':
-            self.resource_obj = pod.get_mds_pods()
-            self.selector = constants.MDS_APP_LABEL
-        if self.resource == 'cephfsplugin':
-            self.resource_obj = pod.get_plugin_pods(
-                interface=constants.CEPHFILESYSTEM
-            )
-            self.selector = constants.CSI_CEPHFSPLUGIN_LABEL
-        if self.resource == 'rbdplugin':
-            self.resource_obj = pod.get_plugin_pods(
-                interface=constants.CEPHBLOCKPOOL
-            )
-            self.selector = constants.CSI_RBDPLUGIN_LABEL
-        if self.resource == 'cephfsplugin_provisioner':
-            self.resource_obj = [pod.plugin_provisioner_leader(
-                interface=constants.CEPHFILESYSTEM
-            )]
-            self.selector = constants.CSI_CEPHFSPLUGIN_PROVISIONER_LABEL
-            resource_count = len(pod.get_cephfsplugin_provisioner_pods())
-        if self.resource == 'rbdplugin_provisioner':
-            self.resource_obj = [pod.plugin_provisioner_leader(
-                interface=constants.CEPHBLOCKPOOL
-            )]
-            self.selector = constants.CSI_RBDPLUGIN_PROVISIONER_LABEL
-            resource_count = len(pod.get_rbdfsplugin_provisioner_pods())
-        if self.resource == 'operator':
-            self.resource_obj = pod.get_operator_pods()
-            self.selector = constants.OPERATOR_LABEL
-
+        resources_dict = {
+            'mgr': [pod.get_mgr_pods(), constants.MGR_APP_LABEL],
+            'mon': [pod.get_mon_pods(), constants.MON_APP_LABEL],
+            'osd': [pod.get_osd_pods(), constants.OSD_APP_LABEL],
+            'mds': [pod.get_mds_pods(), constants.MDS_APP_LABEL],
+            'cephfsplugin': [
+                pod.get_plugin_pods(interface=constants.CEPHFILESYSTEM),
+                constants.CSI_CEPHFSPLUGIN_LABEL
+            ],
+            'rbdplugin': [
+                pod.get_plugin_pods(interface=constants.CEPHBLOCKPOOL),
+                constants.CSI_RBDPLUGIN_LABEL
+            ],
+            'cephfsplugin_provisioner': [
+                [pod.plugin_provisioner_leader(interface=constants.CEPHFILESYSTEM)],
+                constants.CSI_RBDPLUGIN_PROVISIONER_LABEL
+            ],
+            'rbdplugin_provisioner': [
+                [pod.plugin_provisioner_leader(interface=constants.CEPHBLOCKPOOL)],
+                constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
+            ],
+            'operator': [pod.get_operator_pods(), constants.OPERATOR_LABEL]
+        }
+        self.resource_obj = resources_dict[self.resource][0]
```

Comment:
Get pods list only for the resource specified in 'resource' argument. 
functools.partial can be used.

```
from functools import partial
eg: {'mgr': [partial(pod.get_mgr_pods), constants.MGR_APP_LABEL]}  --> This will not call get_mgr_pods.
```
The method can be called here.
```
self.resource_obj = resources_dict[self.resource][0]()
```